device/daita.go

```diff
@@ -0,0 +1,301 @@
+//go:build daita
+// +build daita
+
+package device
+
+import (
+	"time"
+	"unsafe"
+)
+
+// #include <stdio.h>
+// #include <stdlib.h>
+// #include "../cmaybenot/cmaybenot.h"
+// #cgo LDFLAGS: -L${SRCDIR}/../cmaybenot/target/release -l:libcmaybenot.a -lm
+import "C"
+
+type MaybenotDaita struct {
+	events         chan *Event
+	actions        chan *Action
+	maybenot       *C.Maybenot
+	newActionsBuf  []C.MaybenotAction
+	machineActions map[uint64]Action
+	logger         *Logger
+}
+
+type Event struct {
+	// The machine that generated the action that generated this event, if any.
+	Machine uint64
+
+	Peer      NoisePublicKey
+	EventType EventType
+	XmitBytes uint16
+}
+
+type ActionType uint32
+
+const (
```

Comment:
This may be more idiomatic:
```
const (
	ActionTypeCancel ActionType = iota
	ActionTypeInjectPadding
	ActionTypeBlockOutgoing
)
```
`iota` starts at `0` in each new scope where it's used, and the type carries over to subsequent elements unless a different type is specified. By qualifying the names you're also not polluting the namespace.

Reply:
Right now they are defined the same as the corresponding cmaybenot FFI types. We should either make this connection more explicit, or map between the types. Right now we just cast between them.